huaweicloud/services/dew/resource_huaweicloud_csms_secret.go

```diff
@@ -256,49 +255,47 @@ func resourceCsmsSecretUpdate(ctx context.Context, d *schema.ResourceData, meta
 		}
 		_, err = secrets.CreateSecretVersion(client, name, opts)
 		if err != nil {
-			e := fmtp.Errorf("failed to create a new version of CSMS secret: %s", err)
-			mErr = multierror.Append(mErr, e)
+			return diag.Errorf("failed to create a new version of CSMS secret: %s", err)
 		}
 	}
 
 	// Update tags
 	if d.HasChange("tags") {
 		err = utils.UpdateResourceTags(client, d, serviceType, id)
 		if err != nil {
-			e := fmtp.Errorf("failed to update CSMS secret tags: %s", err)
-			mErr = multierror.Append(mErr, e)
+			return diag.Errorf("failed to update CSMS secret tags: %s", err)
 		}
 	}
 
 	if mErr.ErrorOrNil() != nil {
-		return fmtp.DiagErrorf("failed to update CSMS secret: %s", mErr)
+		return diag.Errorf("failed to update CSMS secret: %s", mErr)
 	}
```

Comment:
mErr can be deleted as you just return when an error raised during update.

Reply:
done